### PR TITLE
Reformatted contact shortcuts

### LINKED ISSUE #171: Fix the terrible contact shortcuts row
https://github.com/multacc/multacc/blob/7c86f8940073aa723a9d0a75869e7a350d3a47c3/lib/pages/contacts/contact_details_page.dart#L63-L67

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a `todo` comment in 7c86f8940073aa723a9d0a75869e7a350d3a47c3 when #170 was merged.

### PR DESCRIPTION
**Pull request by DHALL17**
Closes #171

### CODE REVIEW

**Comment by TortoiseWrath** [comment]
Is this important for sprint-3 demo?

File: lib/pages/contacts/contact_details_page.dart [hunk 1]
```diff
@@ -109,35 +134,36 @@ class _ContactDetailsPageState extends State<ContactDetailsPage> {
...
+        crossAxisAlignment: CrossAxisAlignment.center,
         children: <Widget>[
-          _buildContactShortCut(icon: Icons.phone),
-          _buildContactShortCut(icon: Icons.message),
-          _buildContactShortCut(icon: Icons.videocam),
-          _buildContactShortCut(icon: Icons.email),
-          _buildContactShortCut(icon: Icons.dashboard),
+          phoneShortcut != null ? phoneShortcut : _buildShortcutIcon(Icons.phone, "Call", () {}, disabled: true),
+          _buildShortcutIcon(Icons.message, "Text", () {}, disabled: true),
+          emailShortcut != null ? emailShortcut : _buildShortcutIcon(Icons.email, "Email", () {}, disabled: true),
```
Comment (mayank99) [inline lib/pages/contacts/contact_details_page.dart:149]: Need actual `onPressed` implementations

File: lib/pages/contacts/contact_details_page.dart [hunk 2]
```diff
@@ -109,35 +134,36 @@ class _ContactDetailsPageState extends State<ContactDetailsPage> {
...
-      child: ClipOval(
-        child: Material(
-          color: kBackgroundColorLight,
-          child: InkWell(
-            onTap: () {},
-            child: Icon(icon, size: 25),
-          ),
-        ),
-      ),
+  Widget _buildShortcutIcon(IconData icon, String helperText, Function ontap, {bool disabled = false}) {
```
Comment (mayank99) [inline lib/pages/contacts/contact_details_page.dart:155]: This should be named `onPressed`

File: lib/pages/contacts/contact_details_page.dart [hunk 3]
```diff
@@ -109,35 +134,36 @@ class _ContactDetailsPageState extends State<ContactDetailsPage> {
...
+    return InkWell(
```
Comment (mayank99) [inline lib/pages/contacts/contact_details_page.dart:156]: using an icon button or flat button instead of inkwell might make more sense

File: lib/pages/contacts/contact_details_page.dart [hunk 4]
```diff
@@ -59,7 +84,7 @@ class _ContactDetailsPageState extends State<ContactDetailsPage> {
         children: <Widget>[
           Avatars.buildContactAvatar(memoryImage: contact.avatar, radius: 40.0),
           _buildName(),
-          // _buildShortcutsRow(),
+          widget.isProfile ? Container() : _buildShortcutsRow(),
```
Comment (mayank99) [inline lib/pages/contacts/contact_details_page.dart:87]:
```suggestion
          if (!widget.isProfile) _buildShortcutsRow(),
```
No need for empty containers

File: lib/pages/contacts/contact_details_page.dart [hunk 5]
```diff
@@ -16,18 +16,43 @@ import 'package:multacc/pages/contacts/contacts_data.dart';
...
+          emailShortcut = _buildShortcutIcon(Icons.email, "Email", () => (item as EmailItem).launchApp());
+          break;
+        case MultaccItemType.Facebook:
+          return IconButton(
+            icon: Icon(MaterialCommunityIcons.facebook_messenger),
+            onPressed: () => (item as FacebookItem).launchMessenger(),
+          );
+        default:
+          return Text('');
+      }
```
Comment (mayank99) [inline lib/pages/contacts/contact_details_page.dart:52]: Why is this here???

Comment (DHALL17) [reply lib/pages/contacts/contact_details_page.dart:52]: Facebook shouldn't be there. It was just a simple way of having the onTap functions work

Comment (TortoiseWrath) [reply lib/pages/contacts/contact_details_page.dart:52]: why not `_buildShortcutIcon(item.icon, item.type.name, item.launchApp)`?

File: lib/pages/contacts/contact_details_page.dart (hunk 1 above)
Comment (DHALL17) [reply lib/pages/contacts/contact_details_page.dart:149]: Text message doesn't work / didn't want to figure out how to navigate to messaging screen

Comment (TortoiseWrath) [reply lib/pages/contacts/contact_details_page.dart:144]: #108

File: lib/pages/contacts/contact_details_page.dart (hunk 5 above)
Comment (DHALL17) [reply lib/pages/contacts/contact_details_page.dart]: Honestly because That was just how it was being done down the page and I forgot I could do that

**Comment by mayank99** [comment]
Closing this since the `initState` will probably not be fixed
